src/DetailsView/actions/details-view-action-message-creator.ts

```diff
@@ -590,7 +590,21 @@ export class DetailsViewActionMessageCreator extends DevToolActionMessageCreator
         });
     };
 
-    public loadAssessment = (assessmentData: VersionedAssessmentData, tabId: number): void => {
+    public loadAssessment = (
+        assessmentData: VersionedAssessmentData,
+        tabId: number,
+        detailsViewId: string,
+    ): void => {
+        if (assessmentData !== undefined) {
```

Comment:
I'm still not sure this is the right spot for this logic; to be honest, I think this method is already doing too much (it should be sending just a LoadAssessment message, and the details view store should be registering a listener to go to the overview page on assessment load).

To be consistent with our usual flux patterns, the ideal place for business logic related to state changes is in the store responsible for that state. In this case, I think that would mean updating `AssessmentStore.onLoadAssessment` to keep the original `detailsViewId` and ignore the one in the loaded data. This would also be more consistent with other similar logic you've added (particularly, with `AssessmentStore.onResetAllAssessmentsData`)